vscode_extension/README.md

```diff
@@ -1,36 +1,114 @@
+
+![LSP Go to Definition Example](./images/readme/aac_lsp_goto_definition.gif)
+
+#### Diagnostics
+
+AaC provides real-time validation and diagnostics. These diagnostics are driven by the AaC tool's core command `validate`.
+
+![LSP Diagnostics Example](./images/readme/aac_lsp_diagnostics.gif)
+
+#### Document Linking
+
+Document linking allows the IDE to provide clickable navigation links between AaC files.
+
+![LSP Document Linking Example](./images/readme/aac_lsp_document_link.gif)
+
+#### Hover (Preview)
+
+AaC provides a preview of AaC definitions when you hover any reference to them.
 
-### Architecture-as-Code Language Server Protocol (LSP)
-The AaC VSCode extension integrates with the AaC python package to provide IDE language features for AaC files.
+![LSP Hover Example](./images/readme/aac_lsp_hover.gif)
+
+#### Code Completion
+
+AaC supports code completion in order to ease developers' experience when modeling in AaC.
+
+Currently, the support for code completion is limited, but further expansion is planned for the future.
+
+![LSP Code Completion Example](./images/readme/aac_lsp_code_completion.gif)
+
+### Visual Definition Editor (Alpha)
 
-### Visual Definition Editor
 The AaC VSCode extension can leverage the AaC python package's `rest-api` plugin to provide VSCode users with an in-IDE visual editor for AaC definitions.
 
 Before getting started, make sure that you have the AaC `rest-api` plugin running in your project directory.
 ```bash
 aac rest-api --host <hostname|localhost> --port <8000>
 ```
 
+![Visual Editor Example](./images/readme/aac_visual_editor.gif)
+
+## Getting Started
+
+The AaC VSCode extension does not require configuration as long as the `aac`
+Python tool is in your system `PATH`. Nonetheless, there are configuration options
+available, so for detailed instructions on configuring the AaC VSCode Extension,
+refer to [this](https://jondavid-black.github.io/AaC/docs/vscode_extension) page.
+
+### Setting: Server Mode
+
+There are two types of LSP servers available - "IO" and "TCP". Currently, the
+server intended for use by users of the AaC language is the "IO" server (which
+is the default) but, on occasion, it can be beneficial to run the TCP server for
+debugging the tool. In that case, users can switch the "Server Mode" setting to
+"TCP". Otherwise, the recommendation is to keep the "Server Mode" as "IO".
+
+##### TCP
+
+Should you choose to use the "TCP" server mode, you will need to start the LSP
+TCP server separately. After that, there are further settings to configure in
+the VSCode extension.
+
+To start the AaC LSP TCP server, execute the following command:
+
+```bash
+aac start-lsp-tcp
+```
+
+###### Host and Port
+
+When starting the LSP TCP server, it's possible to specify a host and port on
+which to run the server. If you do not specify either the `--host` or the
+`--port` options, the defaults will be used. The default host and port are
```

Comment:
I worded my comment poorly. So going to try this way.

If you do not specify either the `--host` or the `--port` options, the defaults will be used.

Remove both the stand alone sentence and the repeat sentence about default being used.